### PR TITLE
[KYUUBI #4446] Fix connections blocked by Flink insert statements

### PR DESCRIPTION
**Pull request by link3280**
### _Why are the changes needed?_
Flink 1.15 refactors the result fetching of insert statements and now `TableResult.await()` would block till the insert finishes. We could remove this line because the insert results are immediately available as other non-job statements. 

Flink JIRA: https://issues.apache.org/jira/browse/FLINK-24461
Critical changes: https://github.com/apache/flink/pull/17441/files#diff-ec88f0e06d880b53e2f152113ab1a4240a820cbb7248815c5f9ecf9ab4fce4caR108

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Could you please link a Flink JIRA ticket in PR description, and if possible, also add comments in codebase.

**Comment by link3280** [comment]
> Could you please link a Flink JIRA ticket in PR description, and if possible, also add comments in codebase.

Updated as suggested.

**Review by pan3793 — APPROVED** [review]

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala [hunk 1]
```diff
@@ -161,7 +161,8 @@ class ExecuteStatement(
       .build(executor)
     val result = executeOperation.invoke[TableResult](sessionId, operation)
     jobId = result.getJobClient.asScala.map(_.getJobID)
-    result.await()
```
Comment (pan3793) [inline externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:164]: should we reserve it for Flink 1.14? Or there is no side-effect even we remove it?

Comment (link3280) [reply externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:164]: No side-effect for 1.14, the results are generated immediately after the job is submitted.

Comment (pan3793) [reply externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/operation/ExecuteStatement.scala:164]: thanks for confirming, sgtm, pending CI

**Comment by pan3793** [comment]
Thanks, merged to master/1.7/1.6, the fixed version is 1.8.0/1.7.1/1.6.2
